src/ufo.rs

```diff
@@ -406,66 +356,48 @@ impl Font {
             let layer_path = path.join(&layer.path);
-            layer.layer.save(layer_path)?;
+            layer.save(layer_path)?;
         }
 
         Ok(())
     }
 
-    /// Returns a reference to the first layer matching a predicate.
-    /// The predicate takes a `LayerInfo` struct, which includes the layer's
-    /// name and path as well as the layer itself.
-    pub fn find_layer<P>(&self, mut predicate: P) -> Option<&Layer>
-    where
-        P: FnMut(&LayerInfo) -> bool,
-    {
-        self.layers.iter().find(|l| predicate(l)).map(|l| &l.layer)
+    /// Returns a reference to the default layer.
+    pub fn default_layer(&self) -> &Layer {
+        self.layers.default_layer()
     }
 
-    /// Returns a mutable reference to the first layer matching a predicate.
-    /// The predicate takes a `LayerInfo` struct, which includes the layer's
-    /// name and path as well as the layer itself.
-    pub fn find_layer_mut<P>(&mut self, mut predicate: P) -> Option<&mut Layer>
-    where
-        P: FnMut(&LayerInfo) -> bool,
-    {
-        self.layers.iter_mut().find(|l| predicate(l)).map(|l| &mut l.layer)
+    #[deprecated(since = "0.4.0", note = "use default_layer instead")]
+    #[doc(hidden)]
+    pub fn get_default_layer(&self) -> Option<&Layer> {
+        Some(self.default_layer())
     }
 
-    /// Returns a reference to the default layer, if it exists.
-    pub fn get_default_layer(&self) -> Option<&Layer> {
-        self.layers
-            .iter()
-            .find(|l| l.path.file_name() == Some(OsStr::new(DEFAULT_GLYPHS_DIRNAME)))
-            .map(|l| &l.layer)
+    /// Returns a mutable reference to the default layer.
+    pub fn default_layer_mut(&mut self) -> &mut Layer {
+        self.layers.default_layer_mut()
     }
 
-    /// Returns a mutable reference to the default layer, if it exists.
+    #[deprecated(since = "0.4.0", note = "use default_layer instead")]
+    #[doc(hidden)]
     pub fn get_default_layer_mut(&mut self) -> Option<&mut Layer> {
-        self.layers
-            .iter_mut()
-            .find(|l| l.path.file_name() == Some(OsStr::new(DEFAULT_GLYPHS_DIRNAME)))
-            .map(|l| &mut l.layer)
+        Some(self.default_layer_mut())
     }
 
     /// Returns an iterator over all layers in this font object.
-    pub fn iter_layers(&self) -> impl Iterator<Item = &LayerInfo> {
+    pub fn iter_layers(&self) -> impl Iterator<Item = &Layer> {
         self.layers.iter()
     }
 
     /// Returns an iterator over all the glyphs in the default layer.
     pub fn iter_names(&self) -> impl Iterator<Item = GlyphName> + '_ {
-        // this is overly complicated for opaque lifetime reasons, aka 'trust me'
-        self.layers
-            .iter()
-            .filter(|l| l.path.file_name() == Some(OsStr::new(DEFAULT_GLYPHS_DIRNAME)))
-            .flat_map(|l| l.layer.glyphs.keys().cloned())
+        self.layers.default_layer().glyphs.keys().cloned()
     }
 
     //FIXME: support for multiple layers.
```

Comment:
ufoLib2 doesn't support multiple layers either, it's ok to make the user access other layers in question directly.